buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java

```diff
@@ -230,33 +303,22 @@ private void assertNoJarHell(Set<String> jdkJarHellClasses) {
         }
     }
 
-    private void assertNoMissingAndViolations(Set<String> missingClasses, Set<String> violationsClasses) {
-        missingClasses.removeAll(excludes);
-        violationsClasses.removeAll(excludes);
-        String missingText = formatClassList(missingClasses);
-        String violationsText = formatClassList(violationsClasses);
-        if (missingText.isEmpty() && violationsText.isEmpty()) {
-            getLogger().info("Third party audit passed successfully");
-        } else {
-            throw new IllegalStateException(
-                "Audit of third party dependencies failed:\n" +
-                    (missingText.isEmpty() ?  "" : "Missing classes:\n" + missingText) +
-                    (violationsText.isEmpty() ? "" : "Classes with violations:\n" + violationsText)
-            );
+    private void assertNoPointlessExclusions(String specifics, Set<String> excludes, Set<String> problematic) {
+        String notMissing = excludes.stream()
+            .filter(each -> problematic.contains(each) == false)
+            .map(each -> "  * " + each)
+            .collect(Collectors.joining("\n"));
+        if (notMissing.isEmpty() == false) {
+            getLogger().error("Invalid exclusions, following classes " + specifics + ":\n {}", notMissing);
```

Comment:
Maybe use "unnecessary" instead of "invalid"? The latter might imply a syntax error, when instead the exclusion should just be removed

Reply:
changed